src/dds/rs-dds-device-proxy.cpp

```diff
@@ -685,4 +685,14 @@ void dds_device_proxy::update( const void * /*image*/, int /*image_size*/, rs2_u
 }
 
 
+std::vector< sensor_interface * > dds_device_proxy::get_serializable_sensors() const
+{
+    std::vector< sensor_interface * > sensors;
+    auto const n_sensors = get_sensors_count();
+    for( auto i = 0; i < n_sensors; ++i )
+        sensors.push_back( const_cast< sensor_interface * >( &get_sensor( i ) ) );
```

Comment:
Why do we `const_cast` here?
Won't it be better to return `std::vector< const sensor_interface * >`,  or even not declaring this function as cannot change internal members?

Reply:
Well, from memory:
This function is used both by `serialize_json()` (const) and `load_json()` (non-const).
I guess I could split this function into two virtuals, one const and one non-const.